File: C-Sharp/CodilityScenarios/4. Counting Elements/MissingIntegers.cs
using System.Linq;

namespace CodilityScenarios
{
    /// <summary>
    /// Write a function that, given an array A of N integers, returns the 
    /// smallest positive integer(greater than 0) that does not occur in A.
    /// </summary>
    public class MissingIntegers
    {
        public int solution(int[] numericArray)
        {
            int minimalReturnValue = 1;

            // defensive code
            if (numericArray.Length == 0)
            {
                return minimalReturnValue;
            }

            // Handle edge case negative array
            if (numericArray.All(x => x <= 0))
            {
                return minimalReturnValue;
            }

            // Get the potential possible list entries between known min (1) and the max array 
            // value, then find the first (smallest) missing instance
            int returnValue = Enumerable.Range(minimalReturnValue, numericArray.Max())
                                    .Except(numericArray)
                                    .FirstOrDefault();

            // If we have a positive value then return that
            if (returnValue > 0)
            {
                return returnValue;
            }

            if (returnValue < 0)
            {
                return minimalReturnValue;
            }

            // If we hit a this, assume sequential array so return the next max value
            return numericArray.Max() + 1;
        }
    }
}

File: Python/4. Counting Elements/MissingIntegers.py
# Write a function that, given an array A of N integers, returns the 
# smallest positive integer(greater than 0) that does not occur in A.
class MissingIntegers(object):

    def solution(self, numericArray):

        minReturnValue = 1

        # defensive code
        if len(numericArray) == 0:
            return minReturnValue
        
        # check edge case, all negative so next positive has to be 1
        if all(value < 0 for value in numericArray):
            return minReturnValue
        
        # get the known possible values between min (always 1) and max of values provided
        arrayRange = list(range(1, (max(numericArray) +1)))

        # find first number not encountered, default is the next possible value (edge case)
        returnValue = next((value for value in arrayRange if value not in numericArray), (max(numericArray) + 1))

        # return what we know
        if returnValue > 0:
            return returnValue
        
        # further bolt and braces against negative values
        if returnValue < 0:
            return minReturnValue
